mlir/tools/rocmlir-gen/rocmlir-gen.cpp

```diff
@@ -3746,6 +3748,30 @@ int main(int argc, char **argv) {
   llvm::cl::ParseCommandLineOptions(argc, argv,
                                     "MLIR Rock Dialect host generation\n");
 
+  if (!arch.getValue().empty()) {
+    bool archPrefersOCP = (archChip().substr(0, 5) == "gfx12");
```

Comment:
... this isn't the test. gfx950

Can you maybe pull `amdgpu::Chipset` in here?

Reply:
Hmm.  If I contemplate amdgpu::Chipset, I start looking at TargetInfo and ArchInfo, and now wonder if AmdArchInfo::hasFp8ConversionInstrs is an adequate flag to indicate prefers-OCP.  (At the moment only true for gfx12, but there is no gfx950 in AmdArchDb.cpp.)

Reply:
`hasFp8ConversionInsts` isn't the right flag - gfx94x have FP8 conversion instructions. You'll need to add a new flag